pkg/eventemitter/azure_event_grid_handler.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// ******************************* DESCRIPTION ****************************** \\
+// AzureEventGridHandler focuses on emitting the CloudEventSource to Azure Event Grid
+// ************************************************************************** \\
+
+package eventemitter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/messaging"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventgrid/publisher"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	eventingv1alpha1 "github.com/kedacore/keda/v2/apis/eventing/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/eventemitter/eventdata"
+)
+
+type AzureEventGridHandler struct {
+	Endpoint     string
+	Key          string
+	ClusterName  string
+	Client       *publisher.Client
+	logger       logr.Logger
+	activeStatus metav1.ConditionStatus
+}
+
+func NewAzureEventGridHandler(clusterName string, spec *eventingv1alpha1.AzureEventGridSpec, logger logr.Logger) (*AzureEventGridHandler, error) {
+	client, err := publisher.NewClientWithSharedKeyCredential(spec.EndPoint, azcore.NewKeyCredential(spec.Key), nil)
```

Comment:
Should we use `azeventgrid.NewClient()` also to support MSI in the emitter?

Reply:
I'm wondering is it ok to using TriggerAuthentication to provide podidentity here if we want to support MSI. What I concern is that the name "TriggerAuthentication" is for trigger not for emitter. What's your opinion? @JorTurFer @tomkerkhove

Reply:
I think we should and eventually rename it to ScalingAuthentication so that it's no longer related to triggers only.

What do you think about using TriggerAuthentication here, @kedacore/keda-maintainers?

Reply:
Should I create another PR to introduce "TriggerAuthentication" to cloudeventsource?

Reply:
I think that we should think about renaming `TriggerAuthentication` to something like `Authentication` or so, but that's something for V3, until that, I think that we can use `TriggerAuthentication`.

Reply:
Or maybe, we could create another pair of CRD, CloudEventAuthentication and ClusterCloudEventAuthentication. Actually, we won't reuse the parsing logic, so maybe it's a good split for the moment, although I'm a bit afraid about parallel developments about the same stuff if we split them

Reply:
> I think that we should think about renaming TriggerAuthentication to something like Authentication or so, but that's something for V3, until that, I think that we can use TriggerAuthentication.

That is not specifically correct, we can deprecate existing and introduce a new one; but I think that is less relevant here - First we need to decide if we want to do that :)

> Or maybe, we could create another pair of CRD, CloudEventAuthentication and ClusterCloudEventAuthentication. Actually, we won't reuse the parsing logic (at least for the moment), just the MSI stuff, so maybe it's a good split for the moment, although I'm a bit afraid about parallel developments about the same stuff if we split them

For now, that is correct. But I am expecting all the other ones to follow as we onboard more additional capabilities in KEDA so having additional CRDs is a no-go for me personally. I  think we should just re-use what is there and start looking at the above approach

Reply:
> For now, that is correct. But I am expecting all the other ones to follow as we onboard more additional capabilities in KEDA so having additional CRDs is a no-go for me personally. I think we should just re-use what is there and start looking at the above approach

I fully agree with you at that point, a new pair of CRDs it's a no-go (I just suggested some options). I think that currently TriggerAuthentication can be a good starting point, clarifying current limitations. In the future, I'd rename TriggerAuthentication with any other clearest name

Reply:
💯

@SpiritZhou Feel free to use TA / CTA!

Reply:
https://github.com/kedacore/keda/issues/5623